### PR TITLE
Replace CircleCI with GitHub Actions

### PR DESCRIPTION
**Pull request by StanczakDominik**
- [x] I have added a changelog entry for this pull request.

- [x] If adding new functionality, I have added tests and
      docstrings.

- [x] I have fixed any newly failing tests.

### CODE REVIEW

**Comment by StanczakDominik** [comment]
The CircleCI build error here is a sign that we've successfully removed the config :) Upon merging we'll have to

- [ ] head to https://app.circleci.com/settings/project/github/PlasmaPy/PlasmaPy and press Stop Building

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -64,6 +64,11 @@ jobs:
             python: 3.8
             toxenv: py38
 
+          - name: Documentation
+            os: ubuntu-latest
+            python: 3.8
+            toxenv: build_docs
```
Comment (namurphy) [inline .github/workflows/testing.yml:70]: This is really straightforward code.  I thought it was going to be a lot more complicated.  Nice!

**Review by namurphy — COMMENTED** [review]
Looks really close!  I can probably review and approve this pretty quickly once the doc build error is fixed (about pandoc needing to be installed).

File: .github/workflows/testing.yml (hunk 1 above)
Comment (StanczakDominik) [reply .github/workflows/testing.yml:70]: Admittedly I'm not saving any artifacts from this, just having this scream at us about warnings! RTD does the doc output now, so there's no reason to duplicate that.

Comment (StanczakDominik) [reply .github/workflows/testing.yml:70]: Transitioning to `tox` a while ago paid off!

**Comment by StanczakDominik** [comment]
And that's a successful doc build :)

**Review by namurphy — APPROVED** [review]
Goodbye, CircleCI!

**Pull request merged**